rpcserver.go

```diff
@@ -679,13 +679,50 @@ func (r *rpcServer) Start() error {
 
 		go func() {
 			rpcsLog.Infof("gRPC proxy started at %s", lis.Addr())
-			http.Serve(lis, mux)
+			if len(cfg.RestCORS) == 0 {
+				http.Serve(lis, mux)
+			} else {
+				http.Serve(lis, checkCors(mux))
+			}
 		}()
 	}
 
 	return nil
 }
 
+// checkCors allows Cross Origin Resoruce Sharing from configured origins.
+func checkCors(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		origin := r.Header.Get("Origin")
+
+		// Skip CORS check if request did not send origin
+		if origin != "" {
+			// Check if all origins are allowed or requested origin is in list
+			for _, allowedOrigin := range cfg.RestCORS {
```

Comment:
to avoid going over the 80-char line limit, it might help to break this loop out into a helper method

Reply:
Right now it's under 80, longest line is the line 700 with 79 (substituting tabs w/ spaces and then checking count). Would need to restructured this logic a little bit if pulling out into another method, with the return that's in there and all, but think it's still suggested even if it's under 80? 

Is it the tabs making it look longer than 80 in github? I notice the project uses tabs instead of spacing.

Reply:
> Is it the tabs making it look longer than 80 in github? I notice the project uses tabs instead of spacing.

All go code uses tabs since `go fmt` will convert spaces into tabs. Our editors are configured to use 8-space tabs to match the formatting on github, which makes some of these lines 100+ characters.

> Would need to restructured this logic a little bit if pulling out into another method, with the return that's in there and all, but think it's still suggested even if it's under 80?

Yeah i think returning a bool indicating if it's a preflight request would be sufficient

Reply:
Ohhh gotcha, okay yeah it's definitely too long, will work on that!

The contrib guide may need updating to match the 8-space rule: https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#48-additional-style-constraints-on-top-of-gofmt

Reply:
Good catch! That's definitely out of date 😂

Reply:
latest refactoring should put it behind 80 now